packages/medusa/src/services/shipping-option.js

```diff
@@ -475,6 +497,14 @@ class ShippingOptionService extends BaseService {
       }
 
       const optionRepo = manager.getCustomRepository(this.optionRepository_)
+      /**
+       * ERROR: When trying to update with a single requirement (meaning that if there
+       * was previously two, there should now only be one) it fails with the following
+       * Postgres/TypeOrm error "null value in column 'shipping_option_id' of relation
+       * 'shipping_option_requirement' violates not-null constraint". I have tried to
+       * perform a remove on the requirement that should no longer be used, using the
+       * requirementRepo, before saving the option but the error still occurs.
+       */
       const result = await optionRepo.save(option)
```

Comment:
When trying to update with a single requirement (meaning that if there
was previously two, there should now only be one) it fails with the following
Postgres/TypeOrm error "null value in column 'shipping_option_id' of relation
'shipping_option_requirement' violates not-null constraint". I have tried to
perform a remove on the requirement that should no longer be used, using the
requirementRepo, before saving the option but the error still occurs.

any ideas on how I can fix this? @olivermrbl @srindom